src/main/java/org/manifold/compiler/front/ExpressionContextVisitor.java

```diff
@@ -300,11 +297,27 @@ public ExpressionVertex visitNamespacedIdentifier(
   private ExpressionVertex createVariableVertex(VariableIdentifier id) {
     if (ReservedIdentifiers.getInstance()
         .isReservedIdentifier(id)) {
+      // try to get previously added variable vertex to the reserved id
+      if (exprGraph.containsVariable(id)) {
+        return this.getVariableVertex(exprGraph, id);
+      }
       // construct a constant value vertex with the identifier's value
       ConstantValueVertex vReserved = new ConstantValueVertex(exprGraph,
           ReservedIdentifiers.getInstance().getValue(id));
       exprGraph.addVertex(vReserved);
-      return vReserved;
+
+      // create a variable vertex to represent this constant in the graph,
+      // this prevents unnecessary copies of vertices.
+      VariableReferenceVertex v = new VariableReferenceVertex(exprGraph, id);
+      try {
```

Comment:
This try/catch variable creation seems pretty common in this file. Can we just pull variable creation out into a function/generalize it?